<launch>
  <!-- 位姿信息滤波节点 -->
  <!-- <node name="tag_filter" pkg="apriltag_navigation" type="tag_filter.py" output="screen">
    <param name="target_tag_id" value="0" />
    <param name="window_size" value="10" />
  </node> -->
  <arg name="no_static_map" default="false"/>

  <arg name="base_global_planner" default="navfn/NavfnROS"/>
  <!-- <arg name="base_local_planner" default="dwa_local_planner/DWAPlannerROS"/> -->
  <!-- move_base node -->
  <node name="move_base" pkg="move_base" type="move_base" output="screen" respawn="false">
    <!-- <param name="base_global_planner" value="$(arg base_global_planner)"/>
    <param name="base_local_planner" value="$(arg base_local_planner)"/>  

    <rosparam file="$(find tag_navigation)/config/move_base/costmap_common.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find tag_navigation)/config/move_base/costmap_common.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find tag_navigation)/config/move_base/planner.yaml" command="load"/>
    <rosparam file="$(find tag_navigation)/config/move_base/costmap_local.yaml" command="load" ns="local_costmap" />
    <param name="local_costmap/width" value="10.0"/>
    <param name="local_costmap/height" value="10.0"/>
    <rosparam file="$(find tag_navigation)/config/move_base/costmap_global_static.yaml" command="load" ns="global_costmap" unless="$(arg no_static_map)"/> -->
  	<rosparam file="$(find tag_navigation)/config/teb_planner/costmap_common_params.yaml" command="load" ns="global_costmap" />
  	<rosparam file="$(find tag_navigation)/config/teb_planner/costmap_common_params.yaml" command="load" ns="local_costmap" />
  	<rosparam file="$(find tag_navigation)/config/teb_planner/local_costmap_params.yaml" command="load" />
  	<rosparam file="$(find tag_navigation)/config/teb_planner/global_costmap_params.yaml" command="load" />
  	<rosparam file="$(find tag_navigation)/config/teb_planner/teb_local_planner_params.yaml" command="load" />

		<param name="base_global_planner" value="global_planner/GlobalPlanner" />
		<param name="planner_frequency" value="1.0" />
		<param name="planner_patience" value="5.0" />

		<param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS" />
		<param name="controller_frequency" value="5.0" />
		<param name="controller_patience" value="15.0" />

    <param name="clearing_rotation_allowed" value="false" />
  </node> 
  <!-- navigation node -->
  <node name="navigator" pkg="tag_navigation" type="navigator.py" output="screen">
  </node>
</launch>
